src/Disks/IO/WriteBufferFromAzureBlobStorage.cpp

```diff
@@ -18,21 +18,47 @@ namespace ProfileEvents
 namespace DB
 {
 
+struct WriteBufferFromAzureBlobStorage::PartData
+{
+    Memory<> memory;
+    size_t data_size = 0;
+};
+
+IBufferAllocationPolicy::IBufferAllocationPolicyPtr createBufferAllocationPolicy(const AzureObjectStorageSettings & settings)
```

Comment:
Please remove the first `I` from name `IBufferAllocationPolicyPtr` - it's not an interface, it's a pointer.

TBH the first `I` in name `IBufferAllocationPolicy` feels redundant too - yes it has pure virtual functions, but any code outside `BufferAllocationPolicy.cpp` doesn't need to know that, and also call `IBufferAllocationPolicy::create()` looks not quite good with that first `I`.

Reply:
addressed it